spatialos-sdk/src/worker/connection.rs

```diff
@@ -328,30 +325,79 @@ impl Connection for WorkerConnection {
         }
     }
 
-    fn send_command_request(
+    fn send_command_request<C: Component>(
         &mut self,
-        _entity_id: EntityId,
-        _request: CommandRequest,
-        _timeout_millis: Option<u32>,
-        _command_parameters: CommandParameters,
+        entity_id: EntityId,
+        request: C::CommandRequest,
+        timeout_millis: Option<u32>,
+        allow_short_circuit: bool,
     ) -> RequestId<OutgoingCommandRequest> {
-        unimplemented!()
+        let timeout = match timeout_millis {
+            Some(c) => &c,
+            None => ptr::null(),
+        };
+
+        let command_request = Worker_CommandRequest {
+            reserved: ptr::null_mut(),
+            component_id: C::ID,
+            schema_type: ptr::null_mut(),
+            user_handle: component::handle_allocate(request),
+        };
+
+        let params = Worker_CommandParameters {
+            allow_short_circuit: allow_short_circuit as _,
+        };
+
+        let request_id = unsafe {
+            Worker_Connection_SendCommandRequest(
+                self.connection_ptr,
+                entity_id.id,
+                &command_request,
+                0,
+                timeout,
+                &params,
+            )
+        };
+
+        RequestId::new(request_id)
     }
 
-    fn send_command_response(
+    fn send_command_response<C: Component>(
         &mut self,
-        _request_id: RequestId<IncomingCommandRequest>,
-        _response: CommandResponse,
+        request_id: RequestId<IncomingCommandRequest>,
+        response: C::CommandResponse,
     ) {
-        unimplemented!()
+        unsafe {
+            let raw_response = Worker_CommandResponse {
+                reserved: ptr::null_mut(),
+                component_id: C::ID,
+                schema_type: ptr::null_mut(),
+                user_handle: component::handle_allocate(response),
+            };
+
+            Worker_Connection_SendCommandResponse(
+                self.connection_ptr,
+                request_id.id,
+                &raw_response,
+            );
+        }
     }
 
     fn send_command_failure(
         &mut self,
-        _request_id: RequestId<IncomingCommandRequest>,
-        _message: &str,
-    ) {
-        unimplemented!()
+        request_id: RequestId<IncomingCommandRequest>,
+        message: &str,
+    ) -> Result<(), NulError> {
```

Comment:
Why does this return a result? This operation can't fail.

Reply:
Its because of the error propagation of the `CString::new()`..

From https://doc.rust-lang.org/std/ffi/struct.CString.html#errors
> This function will return an error if the supplied bytes contain an internal 0 byte

Reply:
I'm a bit torn here on whether or not it makes sense to expose this error. On the one hand it seems like a bad API if it panics or arbitrarily rejects the user's message for having null bytes in the middle, on the other hand it seems like exposing the `NulError` leaks implementation details about the fact that the is getting converted into a `CString`. For now I went with what seemed like the least surprising option (not panicking unless the user chooses to do so), but I'm open to alternate approaches as well.

Reply:
I'm happy to keep this here for now and we can revisit this as part of #68